framework/src/modules/interoperability/sidechain/commands/cc_update.ts

```diff
@@ -12,18 +12,261 @@
+		if (partnerChainAccount.status === CHAIN_ACTIVE && !isChainLive) {
+			return {
+				status: VerifyStatus.FAIL,
+				error: new Error(`Sending partner chain ${txParams.sendingChainID} is not live.`),
+			};
+		}
+
+		// Section: Liveness Requirement for the First CCU
+		const livenessRequirementFirstCCU = checkLivenessRequirementFirstCCU(
+			partnerChainAccount,
+			txParams,
+		);
+		if (livenessRequirementFirstCCU.error) {
+			return livenessRequirementFirstCCU;
+		}
+		// Section: Certificate and Validators Update Validity
+		const certificateValidity = checkCertificateValidity(partnerChainAccount, txParams.certificate);
+
+		if (certificateValidity.error) {
+			return certificateValidity;
+		}
+
+		// If params contains a non-empty activeValidatorsUpdate
+		const activeValidatorsValidity = checkActiveValidatorsUpdate(txParams);
+		if (activeValidatorsValidity.error) {
+			return activeValidatorsValidity;
+		}
+
+		const partnerChannelStore = context.getStore(transaction.moduleID, STORE_PREFIX_CHANNEL_DATA);
+		const partnerChannelData = await partnerChannelStore.getWithSchema<ChannelData>(
+			partnerChainIDBuffer,
+			channelSchema,
+		);
+		// Section: InboxUpdate Validity
+		const inboxUpdateValidity = checkInboxUpdateValidity(txParams, partnerChannelData);
+		if (inboxUpdateValidity.error) {
+			return inboxUpdateValidity;
+		}
+
+		return {
+			status: VerifyStatus.OK,
+		};
+	}
+
+	public async execute(
+		context: CommandExecuteContext<CrossChainUpdateTransactionParams>,
+	): Promise<void> {
+		const { header, params: txParams } = context;
+		const chainIDBuffer = getIDAsKeyForStore(txParams.sendingChainID);
+		const partnerChainStore = context.getStore(this.moduleID, STORE_PREFIX_CHAIN_DATA);
+		const partnerChainAccount = await partnerChainStore.getWithSchema<ChainAccount>(
+			chainIDBuffer,
+			chainAccountSchema,
+		);
+
+		const decodedCertificate = codec.decode<Certificate>(certificateSchema, txParams.certificate);
+
+		// if the CCU also contains a non-empty inboxUpdate, check the validity of certificate with liveness check
+		checkValidCertificateLiveness(txParams, header, decodedCertificate);
+
+		const partnerValidatorStore = context.getStore(this.moduleID, STORE_PREFIX_CHAIN_VALIDATORS);
+		const partnerValidators = await partnerValidatorStore.getWithSchema<ChainValidators>(
+			chainIDBuffer,
+			chainValidatorsSchema,
+		);
+
+		// Certificate and Validators Update Validity
+		checkCertificateTimestampAndSignature(
+			txParams,
+			partnerValidators,
+			partnerChainAccount,
+			decodedCertificate,
+			header,
+		);
+
+		// If params contains a non-empty activeValidatorsUpdate
+		checkValidatorsHashWithCertificate(txParams, decodedCertificate, partnerValidators);
+
+		// CCM execution
+		const interoperabilityStore = this.getInteroperabilityStore(context.getStore);
+		const decodedCCMs = txParams.inboxUpdate.crossChainMessages.map(ccm => ({
```

Comment:
if CCM decode fails, it should terminate the chain?

Reply:
yes, it should. Done